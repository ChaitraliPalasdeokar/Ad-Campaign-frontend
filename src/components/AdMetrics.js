// import { ShoppingCartCheckout } from "@mui/icons-material";
// import { Card, Typography } from "@mui/material";

// function ShoppingCart() {
//     return (<div>
        
//         <Typography.Title leve={4}>Dashboard</Typography.Title>
//         {/* <Space direction="horizontal"> */}
//             <DashboardCard title={"Title"} />
//         {/* </Space> */}
//     </div>);
// };

// function DashboardCard({ title }) {
//     return (
//         // <Card>
//         //     {/* <Space direction="horizontal"> */}
//         //         <ShoppingCartCheckout />
//         //         <Statistic title="Orders" value={123888888884} />
//         //     {/* </Space> */}
//         // </Card>
//         <DefaultStatisticsCard
//     );
// };

// export default ShoppingCart;