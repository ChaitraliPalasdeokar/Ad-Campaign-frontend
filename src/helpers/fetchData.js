import axios from "axios";

export const fetchData = async () => {
	try {
        const response = await axios.get(
					"http://localhost:5000/api/v1/data?groupby=source&aggregate=attributed_revenue"
				);
		return response.data;
	} catch (error) {
		console.error(error);
	}
};
