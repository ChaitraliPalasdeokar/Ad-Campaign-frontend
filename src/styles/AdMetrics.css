.ad-metrics {
  background-color: #fff;
  border: 1px solid #ccc;
  border-radius: 5px;
  box-shadow: 2px 2px 5px rgba(0, 0, 0, 0.1);
  padding: 20px;
}

.ad-metrics-header {
  font-size: 1.5rem;
  margin-bottom: 10px;
}

.ad-metrics-data {
  display: flex;
  margin-left: 20%;
  margin-right: 20%;
  margin-top: 10px;
  justify-content: space-between;
}

.ad-metrics-item {
  text-align: center;
}

.ad-metrics-label {
  display: block;
  font-size: 1.2rem;
  font-weight: bold;
  margin-bottom: 5px;
}

.ad-metrics-value {
  /* display: block; */
  font-size: 1.5rem;
  /* font-weight: bold; */
}
