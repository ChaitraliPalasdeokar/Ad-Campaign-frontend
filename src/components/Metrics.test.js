import React from "react";
import { fetchData } from "../helpers/fetchData";
import AdMetrics from "./Metrics";
import { act } from "react-dom/test-utils";
import renderer from "react-test-renderer";
import {
	render
} from "@testing-library/react";


jest.mock("../helpers/fetchData", () => ({
	__esModule: true,
	fetchData: jest.fn(),
}));

describe("Metrics component", () => {
	it("should show metrics data", async () => {
		const mockData = {
			attributed_conversions: 83482,
			attributed_revenue: 9493454.6,
			spends: 1720017.57,
		};
		fetchData.mockResolvedValue(mockData);

		const comp = renderer.create(<AdMetrics />);

		expect(comp.toJSON()).toMatchSnapshot();
	});

	it("should make an API call to get metrics data", async () => {
		const mockData = {
			attributed_conversions: 83482,
			attributed_revenue: 9493454.6,
			spends: 1720017.57,
		};
		fetchData.mockResolvedValue(mockData); 

		let container;
		await act(async () => {
			const { container: c } = render(<AdMetrics />);
			container = c;
		});
		expect(fetchData).toHaveBeenCalledWith(
			"data/metrics"
		);
	});
});