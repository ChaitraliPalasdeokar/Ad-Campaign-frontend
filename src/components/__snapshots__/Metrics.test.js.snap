// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Metrics component should show metrics data 1`] = `
<div
  className="ad-metrics"
>
  <h2
    className="ad-metrics-header"
  >
    Advertising Metrics from June to November, 2022
  </h2>
  <div
    className="ad-metrics-data"
  >
    <div
      className="ad-metrics-item"
    >
      <span
        className="ad-metrics-label"
      >
        Total Revenue:
      </span>
      <span
        className="ad-metrics-value"
      />
    </div>
    <div
      className="ad-metrics-item"
    >
      <span
        className="ad-metrics-label"
      >
        Total Conversions:
      </span>
      <span
        className="ad-metrics-value"
      />
    </div>
    <div
      className="ad-metrics-item"
    >
      <span
        className="ad-metrics-label"
      >
        Total Spend:
      </span>
      <span
        className="ad-metrics-value"
      />
    </div>
  </div>
</div>
`;
